test/rpc/test_rpc_schema.py

```diff
@@ -0,0 +1,320 @@
+    (
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            }
+        ],
+        [
+            "latest",
+        ],
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            12345,
+        ],
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "aaaeee",
+        ],
+        [],
+        [
+            {
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "latest",
+        ],
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": "0x1234",
+            },
+            "latest",
+        ],
+        [
+            {
+                "contract_address": 1324,
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "latest",
+        ],
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": ["0x01", "0x02"],
+                "calldata": [],
+            },
+            "latest",
+        ],
+        [
+            "latest",
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": "0x01",
+                "calldata": [],
+            },
+        ],
+    ),
+)
+def test_schema_raises_on_invalid_args(params):
+    """
+    Test params validation with args
+    """
+    resp = rpc_call("starknet_call", params=params)
+
+    assert "error" in resp
+    error = resp["error"]
+
+    assert error["code"] == RpcErrorCode.INVALID_PARAMS.value
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
+    "run_devnet_in_background",
+    [["--disable-rpc-request-validation"]],
+    indirect=True,
+)
+def test_schema_does_not_raise_on_disabled_request_validation():
+    """
+    Test schema validation is disabled by CLI argument
+    """
+    resp = rpc_call(
+        "starknet_call",
+        params={
+            "block_id": "latest",
+            "request": {
+                "contract_address": 1234,
+                "entry_point_selector": -1,
+                "calldata": ["a", "b", "c"],
+            },
+        },
+    )
+
+    # Error will be raised when trying to execute function, but it shouldn't be the INVALID_PARAMS error
+    error = resp["error"]
+    assert error["code"] != RpcErrorCode.INVALID_PARAMS.value
```

Comment:
Looks shaky; we can't predict the error type?

Reply:
Changed so it checks if error code is one of the codes defined as `starknet_call` possible errors. Don't want to hardcode specific error as I believe it could legally throw any of these depending on in which order `starknet_call` performs its internal operations.

Reply:
Why not do assertion on entries in `RpcErrorCode` instead of hardcoding ints?